src/librustc_mir_build/build/matches/simplify.rs

```diff
@@ -40,11 +62,43 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 }
             }
             if !changed {
-                return; // if we were not able to simplify any, done.
+                // Move or-patterns to the end, because they can result in us
+                // creating additional candidates, so we want to test them as
+                // late as possible.
+                candidate
+                    .match_pairs
+                    .sort_by_key(|pair| matches!(*pair.pattern.kind, PatKind::Or { .. }));
+                return false; // if we were not able to simplify any, done.
             }
         }
     }
 
+    fn create_or_subcanidates<'pat>(
```

Comment:
`create_or_subcanidates` is misspelled; it should be `create_or_subcandidates`.

It is also missing a doc-comment. Its something like:
```rust
/// Given candidate whose match pairs are a single or-pattern,
/// creates a fresh candidate for each of its input subpatterns passed via `pat`.
```

right?